src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,55 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidUserDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+
+        if (user.getLogin() == null) {
+            throw new InvalidUserDataException("User login can not be null!");
+        }
+
+        if (user.getPassword() == null) {
+            throw new InvalidUserDataException("User password can not be null!");
+        }
+
+        if (user.getAge() == null) {
+            throw new InvalidUserDataException("User age can not be null!");
+        }
+
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidUserDataException("This username already exist!");
+        }
+
+        if (user.getLogin().length() < 6) {
+            throw new InvalidUserDataException("Username must have at least 6 characters!");
+        }
+
+        if (user.getPassword().length() < 6) {
+            throw new InvalidUserDataException("Password must have at least 6 characters!");
+        }
+
+        if (user.getAge() < 18) {
+            throw new InvalidUserDataException("User must have at least 18 yo!");
+        }
```

Comment:
The check for `user.getAge() < 18` is sufficient to ensure the user is at least 18 years old. The subsequent check for `user.getAge() < 0` is redundant because a negative age would already be caught by the previous check. Consider removing the redundant age check.